kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -403,7 +435,7 @@ def _init_apr_proof(_init_problem: tuple[str, str]) -> APRProof | APRBMCProof:
             contract,
             method,
             save_directory,
-            reinit=reinit,
+            reinit=(method.qualified_name in out_of_date_methods),
```

Comment:
```suggestion
            reinit=(method.qualified_name in out_of_date_methods or reinit),
```

Reply:
If the user specifies to reinit, we should always do so.

Reply:
If the reinit flag set I mark all methods as out of date so we can print that they're out of date.